src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildALibrary.cs

```diff
@@ -390,72 +388,14 @@ public void It_implicitly_defines_compilation_constants_for_the_target_framework
-        [InlineData("android", "2.2", new[] { "ANDROID", "ANDROID2_2" })]
-        [InlineData("windows", "7.0", new[] { "WINDOWS", "WINDOWS7_0" })]
-        public void It_implicitly_defines_compilation_constants_for_the_target_platform(string targetPlatformIdentifier, string targetPlatformVersion, string[] expectedDefines)
-        {
-            var targetFramework = "net5.0";
-            var testAsset = _testAssetsManager
-                .CopyTestAsset("AppWithLibrary", "ImplicitFrameworkConstants", targetFramework)
-                .WithSource()
-                .WithTargetFramework(targetFramework)
-                .WithProjectChanges(project =>
-                {
-                    //  Manually set target plaform properties
-                    var ns = project.Root.Name.Namespace;
-                    var propGroup = new XElement(ns + "PropertyGroup");
-                    project.Root.Add(propGroup);
-
-                    var platformIdentifier = new XElement(ns + "TargetPlatformIdentifier", targetPlatformIdentifier);
-                    propGroup.Add(platformIdentifier);
-                    var platformVersion = new XElement(ns + "TargetPlatformVersion", targetPlatformVersion);
-                    propGroup.Add(platformVersion);
-                    var platformSupported = new XElement(ns + "TargetPlatformSupported", true);
-                    propGroup.Add(platformSupported);
-                    var disableUnnecessaryImplicitFrameworkReferencesForThisTest = new XElement(ns + "DisableImplicitFrameworkReferences", "true");
-                    propGroup.Add(disableUnnecessaryImplicitFrameworkReferencesForThisTest);
-                });
-
-            AssertDefinedConstantsOutput(testAsset, targetFramework, new[] { "NETCOREAPP", "NET", "NET5_0", "NETCOREAPP3_1" }.Concat(expectedDefines).ToArray());
-        }
-
         [Theory]
-        [InlineData(new[] { "1.0", "1.1" }, "ios", "1.1", new[] { "IOS", "IOS1_1", "IOS1_0" })]
-        [InlineData(new[] { "11.11", "12.12", "13.13" }, "android", "12.12", new[] { "ANDROID", "ANDROID11_11", "ANDROID12_12" })]
-        [InlineData(new string[] { /* Use the built in SdkSupportedTargetPlatform items */}, "windows", "10.0.19041.0", new[] { "WINDOWS", "WINDOWS7_0", "WINDOWS8_0", "WINDOWS10_0_17763_0", "WINDOWS10_0_18362_0", "WINDOWS10_0_19041_0" })]
-        public void It_implicitly_defines_compilation_constants_for_the_target_platform_with_backwards_compatibility(string[] sdkSupportedTargetPlatform, string targetPlatformIdentifier, string targetPlatformVersion, string[] expectedDefines)
+        [InlineData(new string[] { }, "ios", "1.1", new[] { "IOS", "IOS1_1" })]
+        [InlineData(new string[] { }, "android", "2.2", new[] { "ANDROID", "ANDROID2_2" })]
+        [InlineData(new string[] { }, "windows", "7.0", new[] { "WINDOWS", "WINDOWS7_0" })]
+        [InlineData(new string[] { }, "windows", "10.0.19041.0", new[] { "WINDOWS", "WINDOWS10_0_19041_0" })]
```

Comment:
We should not define versioned platform constants: `WINDOWS_10_0_19041`, `ANDROID2_2`, etc.  For now, we should just define the versionless constants: `WINDOWS`, `ANDROID`.

We should look at whether we should have versioned platform constants when we look at adding "or greater" constants for the target framework. @terrajobst

Reply:
Sounds good, I've updated to remove these too.